Type errors when using TypeScript 5.4

<!-- If you are contributing this bug fix then please read the instructions in CONTRIBUTING.md -->

**Describe the bug**

Version: 21.0.1

<!-- A description of what the bug is. -->

**To Reproduce**

<!-- Identify the problem and submit some reproduction code. Prune the reproduction code to remove needless details. State the actual behaviour. -->

```json
{
    "compilerOptions": {
        "module": "Node16",
        "strict": true,
    }
}

```

```ts
import * as tsMorph from 'ts-morph';
console.log(tsMorph);

```

```json
{
  "dependencies": {
    "ts-morph": "^21.0.1",
    "typescript": "^5.4.2"
  }
}

```

Run `tsc`:

```console
$ /Users/oliver/Code/typescript-playground/node_modules/.bin/tsc
node_modules/ts-morph/lib/ts-morph.d.ts:5289:127 - error TS2344: Type 'CompilerNodeToWrappedType<InstanceOf<T>["compilerNode"]>' does not satisfy the constraint 'Node<Node>'.
  Type 'SourceFile | ImportDeclaration | ExportDeclaration | ImportEqualsDeclaration | Identifier | ... 198 more ... | Node<...>' is not assignable to type 'Node<Node>'.
    Type 'SourceFile' is not assignable to type 'Node<Node>'.
      The types returned by 'forgetDescendants()' are incompatible between these types.
        Type 'SourceFile' is not assignable to type 'CompilerNodeToWrappedType<InstanceOf<T>["compilerNode"]>'.

5289 export declare function BaseExpressionedNode<T extends Constructor<ExpressionedNodeExtensionType>, TExpression extends Node = CompilerNodeToWrappedType<InstanceOf<T>["compilerNode"]>>(Base: T): Constructor<BaseExpressionedNode<TExpression>> & T;
                                                                                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/ts-morph/lib/ts-morph.d.ts:5289
```

**Expected behavior**

No type errors.

--- a/deno/ts_morph.d.ts
+++ b/deno/ts_morph.d.ts
@@ -3442,7 +3442,7 @@ export declare class Node<NodeType extends ts.Node = ts.Node> {
    */
   forget(): void;
   /** Forgets the descendants of this node. */
-  forgetDescendants(): this;
+  forgetDescendants(): void;
   /**
    * Gets if the compiler node was forgotten.
    *

--- a/deno/ts_morph.js
+++ b/deno/ts_morph.js
@@ -3057,7 +3057,6 @@ class Node {
     forgetDescendants() {
         for (const child of this._getChildrenInCacheIterator())
             child.forget();
-        return this;
     }
     _forgetOnlyThis() {
         if (this.wasForgotten())

--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
     "packages/ts-morph"
   ],
   "devDependencies": {
-    "dprint": "^0.43.2"
+    "dprint": "^0.43.2",
+    "every-ts": "^2.0.0"
   }
 }

--- a/packages/ts-morph/lib/ts-morph.d.ts
+++ b/packages/ts-morph/lib/ts-morph.d.ts
@@ -3442,7 +3442,7 @@ export declare class Node<NodeType extends ts.Node = ts.Node> {
    */
   forget(): void;
   /** Forgets the descendants of this node. */
-  forgetDescendants(): this;
+  forgetDescendants(): void;
   /**
    * Gets if the compiler node was forgotten.
    *

--- a/packages/ts-morph/src/compiler/ast/common/Node.ts
+++ b/packages/ts-morph/src/compiler/ast/common/Node.ts
@@ -119,8 +119,6 @@ export class Node<NodeType extends ts.Node = ts.Node> {
   forgetDescendants() {
     for (const child of this._getChildrenInCacheIterator())
       child.forget();
-
-    return this;
   }
 
   /**

